include/url.h

```diff
@@ -140,6 +140,13 @@ namespace Url
         Url& deparam(const std::unordered_set<std::string>& blacklist);
 
         /**
+         * Put queries and params in sorted order.
+         *
+         * To ensure consistent comparisons, escape should be called beforehand.
+         */
+        Url& canonical();
```

Comment:
I think `sort_query` would be a better name even if it sorts both, but I'd also be fine breaking the methods out separately.